From 11fd715943842ae0be4bf30a9859d784dbbf6ea4 Mon Sep 17 00:00:00 2001
Message-Id: <11fd715943842ae0be4bf30a9859d784dbbf6ea4.1350995715.git.minovotn@redhat.com>
In-Reply-To: <e1fce7d191263c3a20b0768faf927de7aaa2bfa1.1350995715.git.minovotn@redhat.com>
References: <e1fce7d191263c3a20b0768faf927de7aaa2bfa1.1350995715.git.minovotn@redhat.com>
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Thu, 18 Oct 2012 16:22:23 +0200
Subject: [PATCH 3/3] scsi: more fixes to properties for passthrough devices

RH-Author: Paolo Bonzini <pbonzini@redhat.com>
Message-id: <1350577343-17842-4-git-send-email-pbonzini@redhat.com>
Patchwork-id: 43366
O-Subject: [RHEL 6.4 qemu-kvm PATCH 3/3] scsi: more fixes to properties for passthrough devices
Bugzilla: 831102
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Orit Wasserman <owasserm@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>

Commit 0384783 (scsi-block: remove properties that are not relevant for
passthrough, 2012-07-09) removed one property that should have been
left there, "bootindex".

It also did not touch scsi-generic, while it should have.

Fix both problems.

Reported-by: Alexandre DERUMIER <aderumier@odiso.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry-picked from commit 0f1da449ec65403bc5c028eb9618d0ee598a3615)
---
 hw/scsi-disk.c    | 1 +
 hw/scsi-generic.c | 3 ++-
 2 file modificati, 3 inserzioni(+). 1 rimozione(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/scsi-disk.c    | 1 +
 hw/scsi-generic.c | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/hw/scsi-disk.c b/hw/scsi-disk.c
index 9c4a1d8..edf7198 100644
--- a/hw/scsi-disk.c
+++ b/hw/scsi-disk.c
@@ -1978,6 +1978,7 @@ static SCSIDeviceInfo scsi_disk_info[] = {
         .alloc_req    = scsi_block_new_request,
         .qdev.props   = (Property[]) {
             DEFINE_PROP_DRIVE("drive", SCSIDiskState, qdev.conf.bs),
+            DEFINE_PROP_INT32("bootindex", SCSIDiskState, qdev.conf.bootindex, -1),
             DEFINE_PROP_END_OF_LIST(),
         },
 #endif
diff --git a/hw/scsi-generic.c b/hw/scsi-generic.c
index 49a79d9..470815b 100644
--- a/hw/scsi-generic.c
+++ b/hw/scsi-generic.c
@@ -502,7 +502,8 @@ static SCSIDeviceInfo scsi_generic_info = {
     .destroy      = scsi_destroy,
     .alloc_req    = scsi_new_request,
     .qdev.props   = (Property[]) {
-        DEFINE_BLOCK_PROPERTIES(SCSIDevice, conf),
+        DEFINE_PROP_DRIVE("drive", SCSIDevice, conf.bs),
+        DEFINE_PROP_INT32("bootindex", SCSIDevice, conf.bootindex, -1),
         DEFINE_PROP_END_OF_LIST(),
     },
 };
-- 
1.7.11.7

